Source/Scene/TileMapServiceImageryProvider.js

```diff
@@ -367,6 +356,11 @@ TileMapServiceImageryProvider.prototype._metadataSuccess = function (xml) {
 };
 
 TileMapServiceImageryProvider.prototype._metadataFailure = function (error) {
+  // Only reject if the error is from parsing
+  if (error instanceof RuntimeError) {
```

Comment:
Is it possible to move the call to `_matadataFailure` to a different part of the promise chain so that this check isn't needed?

Reply:
I think I can see how this pattern is problematic given that we're having to differentiate for type of error, but semantically, I think this is doing the right thing - where we want to explicitly resolve in the case on an error with `fetchXML`, but not an error within `metadataSuccess`. Given what the requirement is here (i.e. to resolve even if there is a failure), I think this is a case where promise chaining is working against us (where we have to depend on the result of `fetchXML` but any error from it or `metadataSuccess` are both caught in the same block). I saw the `instanceof` pattern used here in https://javascript.info/promise-error-handling, so that's how I initially checked it. I changed it up slightly so that the error type checking is happening in the catch block, where I feel it is better suited than inside the metadataFailure function.

Reply:
> I saw the instanceof pattern used here in https://javascript.info/promise-error-handling, so that's how I initially checked it. 

I appreciate the citation! I think the major different here is that, in linked example, the error type is very specific to the type of error. In our case, I'm concerned that since `Runtime Error` is very generic, it will be easy for this to become prone to bugs when the code changes in the future-- Any new code that throws a `RuntimeError` will go into this branch rather than being handled by `metadataFailure`. Perhaps we can throw a more specific error, or check the error message.

...Or, this becomes a lot cleaner with async/await syntax, which is now available in our toolkit after removing the eslint es5 restrictions:

````js
async function processMetadata() {
   const result = await this._xmlResource.fetchXML();
   try {
       return this._metadataSuccess(result);
    } catch (e) {
       return this._metadataFailure;
    }
}

````

In the above example, `processMetadata` rejects with any error from `fetchXML` when in happens, and otherwise returns the result of `metadataSuccess` or `metadataFailure` accorsingly.